### ISSUE TITLE
Add compilers section to user docs

### ISSUE THREAD

**Issue by timsnyder** [conda-forge/conda-forge.github.io#1686]
### What should be added?

Expand on the content in the user faq, create an end-user specific section that describes how to use the compiler toolchains:
* tl;dr; install `compilers` package and go
* explain the different packages `_impl`, vs `activation scripts` vs `symlinks`
* document the new `conda-gcc-specs` package
* refer to the maintainer's compilers & runtimes section.  Everything there applies 
* stress that recipes added must use the `compiler()` jinja function, that the metapackages and specs are there for end-user convenience and usability.

**Comment by jakirkham** [comment]
Also related issue ( https://github.com/conda-forge/conda-forge.github.io/issues/893 ).

**Comment by prachi237** [comment]
@timsnyder  so basically what I inferred is one has to open a FAQ in faq section of docs that answers the question :  How to use the compiler toolchains? Is it so ? 
The answer had been already mentioned by you ig?
i.e:
tl;dr; install compilers package and go
explain the different packages _impl, vs activation scripts vs symlinks
document the new conda-gcc-specs package
refer to the maintainer's compilers & runtimes section. Everything there applies
stress that recipes added must use the compiler() jinja function, that the metapackages and specs are there for end-user convenience and usability.

**Comment by prachi237** [comment]
@beckermr  @kathatherine  would like to take this issue can u assign me

**Comment by kathatherine** [comment]
I don't have the permissions to assign you officially. @beckermr, do you mind? Go ahead and get started, @prachi237, if you haven't already!

**Comment by jakirkham** [comment]
Done

**Comment by timsnyder** [comment]
@prachi237 , I think that @beckermr and I were talking about creating a new subsection of the user section of the docs that covers all the areas I created the list for.

**Comment by prachi237** [comment]
> @prachi237 , I think that @beckermr and I were talking about creating a new subsection of the user section of the docs that covers all the areas I created the list for.

Ahaha got it ! I Will try to implement it :)
